File: AndroidRxJavaWithRetrofit2/app/src/main/java/com/blackbirds/shakil/androidrxjavawithretrofit2/Network/APIService.java
package com.blackbirds.shakil.androidrxjavawithretrofit2.Network;

import com.blackbirds.shakil.androidrxjavawithretrofit2.Model.PostModel;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.http.GET;

public interface APIService {
    @GET("posts")
    Observable<List<PostModel>> getPostsList();
}

File: KotlinRxJavaWithRetrofit2/app/src/main/java/com/blackbirds/shakil/kotlinrxjavawithretrofit2/Network/APIService.kt
package com.blackbirds.shakil.kotlinrxjavawithretrofit2.Network

import com.blackbirds.shakil.kotlinrxjavawithretrofit2.Model.PostModel
import io.reactivex.Observable
import retrofit2.http.GET

interface APIService {
    @GET("posts")
    fun postsList(): Observable<List<PostModel>>
}